grunt_tasks/plugin.js

```diff
@@ -61,6 +61,168 @@ module.exports = function (grunt) {
     });
 
     /**
+     * Collect dependencies from a package.json-like object.
+     */
+    function getDependencies(obj, fields, deps) {
+        deps = deps || {};
+        fields = fields || ['dependencies'];
+
+        _.each(fields, function (field) {
+            _.extend(deps, obj[field]);
+        });
+        return deps;
+    }
+
+    /**
+     * Run npm install in the given prefix directory.
+     *
+     * @param {object} [deps]
+     *   A dependency -> version mapping in the same syntax provided
+     *   in a package.json file.
+     *
+     * @param {string} [prefix]
+     *   The `--prefix` argument to the npm command.
+     */
+    function npmInstall(deps, prefix) {
+        var args = [];
+
+        if (prefix) {
+            args = ['--prefix', prefix];
+        }
+        args = args.concat(['--color=always', 'install', '--no-save']);
+
+        if (deps) {
+            args = args.concat(
+                _.map(deps, function (version, dep) {
+                    return dep + '@' + version;
+                })
+            );
+        }
+
+        var child = child_process.spawnSync('npm', args, {
+            stdio: 'inherit'
+        });
+
+        if (child.status !== 0) {
+            throw new Error(`"npm ${args.join(' ')}" failed`);
```

Comment:
I'm run into a case where throwing a new Error inside one of these grunt modules does not actually cause a build failure, but I can't remember if it was at config time or task run time. Does this cause the build to fail? If not, `grunt.fail()` will work.